### PR TITLE
Eng 920 implement operator details page

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
Created:
- Log Viewer Component
- Multi-File Viewer Component
- Operator Details Page

Currently has duplicate code with metrics details page but that will be fixed in a later refactor.

## Related issue number (if any)
ENG-920

## Loom demo (if any)
operator_details_no_logs
operator_details_logs

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/workflows/edges/AqueductBezier.tsx
Comment (eunice-chan) [inline src/ui/common/src/components/workflows/edges/AqueductBezier.tsx:31]: Linting gave me the following error so I removed the viewBox:
`Invalid property 'viewBox' found on tag 'marker', but it is only allowed on: svg  `

File: src/ui/common/src/components/pages/metric/id/index.tsx
Comment (eunice-chan) [inline src/ui/common/src/components/pages/metric/id/index.tsx:240]: Linter complained: `Missing "key" prop for element in iterator`

File: src/ui/common/src/components/workflows/edges/AqueductBezier.tsx
Comment (agiron123) [reply src/ui/common/src/components/workflows/edges/AqueductBezier.tsx:31]: This is strange that this error is just now showing up. 

Could you please file a Linear task for this and reply here so that we can investigate this issue later?

File: src/ui/common/src/components/pages/metric/id/index.tsx
Comment (agiron123) [reply src/ui/common/src/components/pages/metric/id/index.tsx:240]: Yeah, totally expected if we're going to be iterating. As long as the artifactId is uuid or something unique, we're good to go.

File: src/ui/common/package.json
Comment (agiron123) [inline src/ui/common/package.json:109]: Could you please move monaco and zip.js to peerDependencies like the rest of the dependencies that we have in common?

Comment (agiron123) [reply src/ui/common/package.json:109]: After done, I'd do a make clean from ui directory and re-install to make sure everything is good.

File: src/ui/common/src/components/DataPreviewer/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/DataPreviewer/index.tsx:23]: kinda strange that your linter is picking this up. I just worked on this file yesterday and linted before pushing. Let me verify some things on my end and see if there's any issues.

Comment (agiron123) [reply src/ui/common/src/components/DataPreviewer/index.tsx:23]: okay, looks like this is correct. ran lint and this file is being changed as well :)
```
diff --git a/src/ui/common/src/components/DataPreviewer/index.tsx b/src/ui/common/src/components/DataPreviewer/index.tsx
index 1df3ba07..68d84c99 100644
--- a/src/ui/common/src/components/DataPreviewer/index.tsx
+++ b/src/ui/common/src/components/DataPreviewer/index.tsx
@@ -20,7 +20,11 @@ type Props = {
   dataTableHeight?: string;
 };
 
-export const DataPreviewer: React.FC<Props> = ({ previewData, error, dataTableHeight }) => {
+export const DataPreviewer: React.FC<Props> = ({
+  previewData,
+  error,
+  dataTableHeight,
+}) => {
   if (!previewData) {
     return null;
   }
@@ -167,4 +171,4 @@ export const DataPreviewer: React.FC<Props> = ({ previewData, error, dataTableHe
   );
 };
 
-export default DataPreviewer;
\ No newline at end of file
+export default DataPreviewer;

```

File: src/ui/common/src/components/LogViewer/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/LogViewer/index.tsx:55]: Bit of a nitpick, but could you make 'No output to display.' a const please?
```
const noOutputMessage = 'No output to display.'
```

Same for the no errors message and no logs message

File: src/ui/common/src/components/MultiFileViewer/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/MultiFileViewer/index.tsx:136]: I can't comment on the file below because it's empty (silly Github),
but could you remove the empty multi-file-viewer-styles.module.css file please?

File: src/ui/common/src/components/workflows/edges/AqueductStraight.tsx
Comment (agiron123) [inline src/ui/common/src/components/workflows/edges/AqueductStraight.tsx:22]: Same as above.

File: src/ui/common/src/components/workflows/edges/AqueductQuadratic.tsx
Comment (agiron123) [inline src/ui/common/src/components/workflows/edges/AqueductQuadratic.tsx:28]: Instead of removing this can you comment it out and leave a TODO comment to investigate later? I'm curious about this as I'm not getting this linter error on my end and want to investigate further later.

File: src/ui/common/src/components/workflows/SideSheets/OperatorResultsSideSheet.tsx
Comment (agiron123) [inline src/ui/common/src/components/workflows/SideSheets/OperatorResultsSideSheet.tsx:25]: Could you post a screenshot of the OperatorResultsSideSheet with this new log viewer component please? Curious to see how it looks in the side sheet.

**Comment by agiron123** [comment]
Nice work! This is awesome. Left some code comments on things that I had small questions/concerns about.

From a design standpoint, I have a couple questions:
1. There's not a lot of visual separation between the Logs and Code Preview section. Could you add a border around the tab view of the logs section? Basically wrap the Tabs component and it's children with a box and add a 1 or 2 px border around with an 8px borderRadius. Try to use the same gray color as that of the tab view component.
2. Same as above for the Code Preview section.
3. Just curious, but does our editor support syntax highlighting in Python? Are we able to use another theme (say vs code dark) if we wish? I'd love to experiment with using other color themes, but i'm not so sure if we're able to or what work that would entail.

File: src/ui/common/src/components/workflows/edges/AqueductBezier.tsx
Comment (agiron123) [reply src/ui/common/src/components/workflows/edges/AqueductBezier.tsx:31]: Same as my other comments on this viewBox issue. Could you please comment out this line instead of removing and add a TODO: comment to investigate further?

A linear task link in the TODO comment would be awesome.

File: src/ui/common/src/components/workflows/SideSheets/OperatorResultsSideSheet.tsx
Comment (eunice-chan) [reply src/ui/common/src/components/workflows/SideSheets/OperatorResultsSideSheet.tsx:25]: image

File: src/ui/common/src/components/workflows/edges/AqueductBezier.tsx
Comment (eunice-chan) [reply src/ui/common/src/components/workflows/edges/AqueductBezier.tsx:31]: Here is the linear task: https://linear.app/aqueducthq/issue/ENG-1739/look-into-linter-complaint-invalid-property-viewbox

**Comment by eunice-chan** [comment]
@agiron123 

Yes, the editor support syntax highlighting in Python.
The default themes are `light` and `vs-dark` although we can define custom themes as well.

Here is dark theme with the spacing changes you've suggested: 
image

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (likawind) [inline src/ui/common/src/components/pages/function/id/index.tsx:69]: I moved this mapping to https://github.com/aqueducthq/aqueduct/pull/507

**Review by likawind — APPROVED** [review]
This is awesome! The page and file viewer is really cool! Let's coordinate the merge so that we could reuse the artifact-to-icon map.

After this is merged, I think we can have a follow up task with some polish around the file viewer. We could refer to some external packages like https://www.npmjs.com/package/react-file-treeview (or @agiron123 may have better ideas)

**Comment by vsreekanti** [comment]
@eunice-chan, this is awesome! Before merging, could you please record a demo of the latest version of the code? I don't want to repeat any feedback that Wei and Andre already provided, but I wasn't sure exactly what had/hadn't been changed. Thank you!

**Comment by agiron123** [comment]
@eunice-chan good work, let's go with the light theme here. Borders around the sections look great.

File: src/ui/common/src/components/LogViewer/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/LogViewer/index.tsx:20]: remove log

File: src/ui/common/src/components/MultiFileViewer/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/MultiFileViewer/index.tsx:29]: What's going on here with the media query and event listener?
Was this code taken from an example on the mui.com website?

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/pages/function/id/index.tsx:52]: Should this be named OperatorDetailsPage instead of FunctionDetailsPage? It's the title of your PR :) Either one should be fine, but our code base has more usage of "Operator" than "Function"  in terms of naming.

cc @vsreekanti and @likawind for second opinion here.

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (agiron123) [reply src/ui/common/src/components/pages/function/id/index.tsx:62]: Instead of just passing in the object here, you should probably define a type above and pass that into useState.

e.g.

```typescript
const FilePreview = {
  path: '',
  language: 'plaintext',
  content: ''
};

const [files, setFiles] = useState<FilePreview>(null);
```

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (agiron123) [inline src/ui/common/src/components/pages/function/id/index.tsx:120]: @likawind should we be using your new API route for getting the file results here?

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (likawind) [reply src/ui/common/src/components/pages/function/id/index.tsx:52]: Yeah..If this `index.tsx` is what we will use for `operator/{id}` url we should call it `OperatorDetailPage` and rename the directory as well

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (likawind) [reply src/ui/common/src/components/pages/function/id/index.tsx:120]: I think we should do all route migration separately. We can get the functional part merged first.

For operator file in particular, we don't have another new route to get files yet (we assume file could be large so it's not included in the new route's response), so for file it's good to use the endpoint underlying `handleGetOperatorResults` now.

File: src/ui/common/src/components/pages/function/id/index.tsx
Comment (vsreekanti) [reply src/ui/common/src/components/pages/function/id/index.tsx:52]: Agreed that operator is more general, and we should be consistent.

**Comment by eunice-chan** [comment]
Merged to unblock Wei. Will be making a followup PR that will do the following:
- update API usage to the new one Wei made
- reuse components shared with the other detail pages
- resolve the comments made in this PR
- resolve any other cosmetic bugs that may've been introduced in this PR (notably, the workflows page's error bar should be dealt with by/in the PR)

**Pull request merged**
